packages/wmr/src/lib/prerender.js

```diff
@@ -96,17 +96,41 @@ async function workerCode({ cwd, out, publicPath, customRoutes }) {
 	let head = { lang: '', title: '', elements: new Set() };
 	globalThis.wmr = { ssr: { head } };
 
+	async function localFetcher(url) {
+		const text = () => fs.readFile(`${out}/${String(url).replace(/^\//, '')}`, 'utf-8');
+		return { text, json: () => text().then(JSON.parse) };
+	}
+
+	const pattern = /^\//;
 	if (globalThis.fetch === undefined) {
+		// When Node.js version under 17, native fetch API undefined
+		// So it will give a warning when the syntax to retrieve the file and the url is passed
 		// @ts-ignore
 		globalThis.fetch = async url => {
-			const pattern = /^\//;
-			const isLocalFile = pattern.test(String(url));
-			if (isLocalFile) {
-				const text = () => fs.readFile(`${out}/${String(url).replace(/^\//, '')}`, 'utf-8');
-				return { text, json: () => text().then(JSON.parse) };
+			if (pattern.test(String(url))) {
+				return localFetcher(url);
+			}
+
+			console.warn(`fetch is not defined in Node.js version under 17, please upgrade to Node.js version 18 later`);
+		};
+	} else {
+		// At that time, implement using reassignment to avoid entering an infinite loop.
+		const fetcher = fetch;
+		// @ts-ignore
+		delete globalThis.fetch;
+
+		// When Node.js version 18 later, native fetch API is defined
+		// Override with own implementation if you want to retrieve local files in Node.js 18 later
+		// ref https://github.com/preactjs/wmr/pull/935#discussion_r977168334
+		// @ts-ignore
+		globalThis.fetch = async (url, options) => {
+			if (pattern.test(String(url))) {
+				return localFetcher(url);
 			}
 
-			console.warn(`fetch is not implemented in Node.js, please upgrade to Node.js 18 or above`);
+			// When fetching an external resource, it returns the native fetch API
+			// though `fetcher` function is an alias created to avoid infinite loops
+			return fetcher(url, options);
```

Comment:
overrided globalThis.fetch with our own implementation.
ref https://github.com/preactjs/wmr/pull/935#discussion_r977168334

Reply:
The `fetcher` function is an alias function defined to avoid infinite loops and is equivalent to the native `fetch` function.

Reply:
When Node.js version 18 later, I addressed this by writing conditional expressions for both `fetch` is defined and it is undefined to override. 
I also wrote corresponding tests.

Reply:
Local fetching should now be automated. If you expect a different behavior from the fetch that wmr itself is overriding, you should be able to solve this by writing the following code.

```js
export async function prerender(vnode: VNode) {
    globalThis.funkyFetch = ...
}

const fetcher = typeof window === 'undefined' ? funkyFetch : fetch;
fetcher(...);
```